src/core_modules/capture-core/components/MapCoordinates/MapCoordinatesModal.js

```diff
@@ -0,0 +1,153 @@
+    const [position, setPosition] = useState(null);
+    const [coordinates, setCoordinates] = useState(null);
+
+    const onHandleMapClicked = (mapCoordinates) => {
+        if (type === dataElementTypes.COORDINATE) {
+            const { lat, lng } = mapCoordinates.latlng;
+            const newPosition: [number, number] = [lat, lng];
+            // $FlowFixMe
+            setPosition(newPosition);
+        }
+    };
+
+    const onMapPolygonCreated = (e: any) => {
+        const polygonCoordinates = e.layer.toGeoJSON().geometry.coordinates;
+        setCoordinates(polygonCoordinates);
+    };
+
+    const onMapPolygonEdited = (e: any) => {
+        const polygonCoordinates = e.layers.getLayers()[0].toGeoJSON().geometry.coordinates;
+        setCoordinates(polygonCoordinates);
+    };
+    const onMapPolygonDelete = () => {
+        setCoordinates(null);
+    };
+
+    const renderMap = () => (<Map
+        center={center}
+        zoom={13}
+        ref={(ref) => {
+            if (ref?.leafletElement) {
+                setTimeout(() => { ref.leafletElement.invalidateSize(); }, 250);
+            }
+        }}
+        className={classes.map}
+        onClick={onHandleMapClicked}
+    >
+        <TileLayer
+            url="http://{s}.tile.osm.org/{z}/{x}/{y}.png"
+            attribution="&copy; <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors"
+        />
+        {type === dataElementTypes.POLYGON && <FeatureGroup>
+            <EditControl
+                position="topright"
+                onEdited={onMapPolygonEdited}
+                onCreated={onMapPolygonCreated}
+                onDeleted={onMapPolygonDelete}
+                draw={{
+                    rectangle: false,
+                    polyline: false,
+                    circle: false,
+                    marker: false,
+                    circlemarker: false,
+                }}
+                edit={{
+                    remove: false,
+                }}
+            />
+        </FeatureGroup>}
+        {type === dataElementTypes.COORDINATE && position && <Marker position={position} />}
+    </Map>);
+
+    const getTitle = () => {
+        switch (type) {
+        case dataElementTypes.COORDINATE:
+            return i18n.t('coordinates');
+        case dataElementTypes.POLYGON:
+            return i18n.t('area');
+        default:
+            log.error(`${type} is not handled`);
+            return '';
+        }
+    };
+
+    const renderActions = () => (<ButtonStrip end>
+        <Button
+            onClick={() => {
+                setOpen(false);
+            }}
+            secondary
+        >
+            {i18n.t('Cancel')}
+        </Button>
+        <Button
+            onClick={() => {
+                if (position ?? coordinates) {
+                    const convertedCoordinates = convertToServerCoordinates(position ?? coordinates, type);
+                    // $FlowFixMe
```

Comment:
Hey, 
How about adding return types to the `convertToServerCoordinates` function and getting rid of the `// $FlowFixMe` comment? This worked when I tested it: 
```
const convertToServerCoordinates =
(coordinates?: Array<[number, number]> | null, type: string): ?[number, number] | ?Array<[number, number]> => {
    if (!coordinates) { return null; }
    switch (type) {
    case dataElementTypes.COORDINATE:
        return Array<[number, number]>([coordinates[1], coordinates[0]]);
    case dataElementTypes.POLYGON:
        return [coordinates[0]];
    default:
        return coordinates;
    }
};
```
WDYT?